File: 0014 Longest Common Prefix/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int n = strs.size();
        if (n == 0)  return "";
        
        string prefix = strs[0];
        for (int i = 0; i < prefix.size(); i ++)
        {
            for (int j = 0; j < n; j ++)
            {
                if (i > strs[j].size() - 1 || prefix[i] != strs[j][i])
                    return prefix.substr(0, i);
            }
        }
        return prefix;
    }
};


int main(){
    Solution s;
    vector<string> vec = {"flower","flow","flight"};
    assert(s.longestCommonPrefix(vec) == "fl");
    return 0;
}

File: 0014 Longest Common Prefix/solution.py
#! python3

class Solution:
    def longestCommonPrefix(self, str):
        if not str:
            return ""
        res = str[0]
        for i in range(1, len(str)):
            while str[i].find(res) != 0:
                res = res[0: len(res) - 1]
            if not res:
                return ""
            
        return res


if __name__ == "__main__":
    strs = ["flower","flow","flight"]
    print(Solution().longestCommonPrefix(strs) == "fl")